server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/ClusterBalanceStats.java

```diff
@@ -158,7 +171,27 @@ private static NodeStats createFrom(RoutingNode routingNode, Metadata metadata,
                 totalShardSize += indexMetadata.getForecastedShardSizeInBytes().orElse(0);
             }
 
-            return new NodeStats(routingNode.size(), totalWriteLoad, totalShardSize);
+            return new NodeBalanceStats(routingNode.size(), totalWriteLoad, totalShardSize);
+        }
+
+        public static NodeBalanceStats readFrom(StreamInput in) throws IOException {
+            return new NodeBalanceStats(in.readInt(), in.readDouble(), in.readLong());
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeInt(shards);
+            out.writeDouble(forecastedWriteLoad);
+            out.writeLong(forecastedShardSize);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            return builder.startObject()
+                .field("shard_count", shards)
+                .field("forecasted_write_load", forecastedWriteLoad)
+                .humanReadableField("forecasted_disk_usage_bytes", "forecasted_disk_usage", ByteSizeValue.ofBytes(forecastedShardSize))
```

Comment:
I think it would be beneficial to also add actual disk usage here.
I think there is a confusion between forecasted disk usage and actual disk usage, may be adding both here would help to highlight the difference